src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -234,13 +235,94 @@ public AndroidTzData()
                     {
                         _tzFileDir = tzFileDir;
                         _tzFilePath = tzFilePath;
+                        _isFiltered = false;
                         return;
                     }
                 }
 
                 throw new TimeZoneNotFoundException(SR.TimeZoneNotFound_ValidTimeZoneFileMissing);
             }
 
+            // On some versions of Android, the tzdata file may still contain backward timezone ids.
+            // We attempt to use tzlookup.xml, which is available on some versions of Android to help
+            // validate non-backward timezone ids
+            // tzlookup.xml is an autogenerated file that contains timezone ids in this form:
+            //
+            // <timezones ianaversion="2019b">
+            //   <countryzones>
+            //     <country code="au" default="Australia/Sydney" everutc="n">
+            //       <id alts="Australia/ACT,Australia/Canberra,Australia/NSW">Australia/Sydney</id>
+            //       ...
+            //       ...
+            //       <id>Australia/Eucla</id>
+            //     </country>
+            //     <country ...>
+            //       ...
+            //       ...
+            //       ...
+            //     </country>
+            //   </countryzones>
+            // </timezones>
+            //
+            // Once the timezone cache is populated with the IDs, we reference tzlookup id tags and
+            // remove IDs and their associated information (byteoffsets and lengths) from the current
+            // AndroidTzData instance.
+            private void filterBackwardIDs(string lookupPath)
```

Comment:
```suggestion
            private void FilterBackwardIDs(string lookupPath)
```
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/docs/coding-guidelines/coding-style.md
> 13. We use PascalCasing for all method names, including local functions.

Reply:
Thank you :)